boa/src/builtins/json/mod.rs

```diff
@@ -63,8 +67,35 @@ pub fn parse(_: &mut Value, args: &[Value], _: &mut Interpreter) -> ResultValue
 /// [spec]: https://tc39.es/ecma262/#sec-json.stringify
 /// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify
 pub fn stringify(_: &mut Value, args: &[Value], _: &mut Interpreter) -> ResultValue {
-    let obj = args.get(0).expect("cannot get argument for JSON.stringify");
-    let json = obj.to_json().to_string();
+    let object = args.get(0).expect("cannot get argument for JSON.stringify");
+    let object_to_return = Value::new_object(None);
+    if let Some(arg) = args.get(1) {
+        match arg.data() {
```

Comment:
The warning here is just that it would be better to do:
```rust
if let ValueData::Object(ref obj) = arg.data() {
    // Deal with it
} else {
   unimplemented!("...");
}
```

Reply:
:heavy_check_mark: